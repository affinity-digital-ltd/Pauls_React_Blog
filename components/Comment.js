import { Component } from 'react'

// import moment from 'moment'
import { MarkdownPreview } from 'react-marked-markdown'

class Comment extends Component {
  render () {
    const { comment } = this.props

    return (
      <div className='c-comment'>
        <div className='c-comment__author'>
          <img className='c-comment__img' src={comment.image} />
          <h4 className='c-comment__author-name'>{comment.authorName}</h4>
          <time className='c-comment__created-date'>Created on: {comment.createdAt}</time>
        </div>
        <div className='c-comment__body'>
          <MarkdownPreview value={comment.body} />
        </div>
      </div>
    )
  }
}

export default Comment
