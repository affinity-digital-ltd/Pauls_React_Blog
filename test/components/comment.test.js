import Comment from '../../components/Comment'

describe('<Comment />', () => {
  it('it renders correctly', () => {
    const comment = {
      authorName: 'authorName'
    }
    
    const component = shallow(
      <Comment comment={comment} />
    )

    expect(component).toMatchSnapshot()
  })
})
