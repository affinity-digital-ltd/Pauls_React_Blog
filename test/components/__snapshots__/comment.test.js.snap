// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Comment /> it renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Comment
    comment={
      Object {
        "authorName": "authorName",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="c-comment__author"
        >
          <img
            alt="A Ruby logo"
            className="c-comment__img"
            src="../static/images/ruby.jpg"
          />
          <h4
            className="c-comment__author-name"
          >
            authorName
          </h4>
          <time
            className="c-comment__created-date"
          >
            Created on: 
            Dec 14, 2018
          </time>
        </div>,
        <div
          className="c-comment__body"
        >
          <MarkdownPreview
            value=""
          />
        </div>,
      ],
      "className": "c-comment",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="A Ruby logo"
              className="c-comment__img"
              src="../static/images/ruby.jpg"
            />,
            <h4
              className="c-comment__author-name"
            >
              authorName
            </h4>,
            <time
              className="c-comment__created-date"
            >
              Created on: 
              Dec 14, 2018
            </time>,
          ],
          "className": "c-comment__author",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "A Ruby logo",
              "className": "c-comment__img",
              "src": "../static/images/ruby.jpg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "authorName",
              "className": "c-comment__author-name",
            },
            "ref": null,
            "rendered": "authorName",
            "type": "h4",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Created on: ",
                "Dec 14, 2018",
              ],
              "className": "c-comment__created-date",
            },
            "ref": null,
            "rendered": Array [
              "Created on: ",
              "Dec 14, 2018",
            ],
            "type": "time",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <MarkdownPreview
            value=""
          />,
          "className": "c-comment__body",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="c-comment__author"
          >
            <img
              alt="A Ruby logo"
              className="c-comment__img"
              src="../static/images/ruby.jpg"
            />
            <h4
              className="c-comment__author-name"
            >
              authorName
            </h4>
            <time
              className="c-comment__created-date"
            >
              Created on: 
              Dec 14, 2018
            </time>
          </div>,
          <div
            className="c-comment__body"
          >
            <MarkdownPreview
              value=""
            />
          </div>,
        ],
        "className": "c-comment",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="A Ruby logo"
                className="c-comment__img"
                src="../static/images/ruby.jpg"
              />,
              <h4
                className="c-comment__author-name"
              >
                authorName
              </h4>,
              <time
                className="c-comment__created-date"
              >
                Created on: 
                Dec 14, 2018
              </time>,
            ],
            "className": "c-comment__author",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "A Ruby logo",
                "className": "c-comment__img",
                "src": "../static/images/ruby.jpg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "authorName",
                "className": "c-comment__author-name",
              },
              "ref": null,
              "rendered": "authorName",
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Created on: ",
                  "Dec 14, 2018",
                ],
                "className": "c-comment__created-date",
              },
              "ref": null,
              "rendered": Array [
                "Created on: ",
                "Dec 14, 2018",
              ],
              "type": "time",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <MarkdownPreview
              value=""
            />,
            "className": "c-comment__body",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
